lang/c++/impl/avrogencpp.cc

```diff
@@ -503,35 +520,21 @@ void CodeGen::generateEnumTraits(const NodePtr& n)
 {
 	string dname = decorate(n->name());
 	string fn = fullname(dname);
-	size_t c = n->names();
-	string first;
-	string last;
-	if (!ns_.empty())
-	{
-		first = ns_;
-		first += "::";
-		first += n->nameAt(0);
-
-		last = ns_;
-		last += "::";
-		last += n->nameAt(c-1);
-	} else {
-		first = n->nameAt(0);
-		last = n->nameAt(c-1);
-	}
+	string last = n->nameAt(n->names() - 1);
+
 	os_ << "template<> struct codec_traits<" << fn << "> {\n"
 		<< "    static void encode(Encoder& e, " << fn << " v) {\n"
-		<< "		if (v < "  << first << " || v > " << last << ")\n"
+		<< "		if (v > " << fn << "::" << last << ")\n"
```

Comment:
Here's the updated enum traits code using the last enum symbol rather than defining a `MAX` value.

Reply:
That works.  :)